tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -251,6 +251,23 @@ StorageManager::array_open_for_reads(Array* array) {
           *array->encryption_key());
   RETURN_NOT_OK_TUPLE(st, nullopt, nullopt, nullopt);
 
+  // If building experimentally, this library should not be able to
+  // read from newer-versioned arrays
+  if constexpr (is_experimental_build) {
```

Comment:
Should this check be introduced for experimental builds only or for all of them? @ihnorton can you clarify the requirement?

Reply:
Yes, @bekadavis9 this part should be unconditional. We want to block reading from newer format versions unconditionally (because it is unsupported), but allow reading older versions; the experimental-build format version will always be greater than any release-build format version.